logprep/framework/rule_tree/rule_parser.py

```diff
@@ -126,15 +130,36 @@ def _add_exists_filter(parsed_rules: list):
         for parsed_rule in parsed_rules:
             temp_parsed_rule = parsed_rule.copy()
             skipped_counter = 0
-
-            for segment_index, segment in enumerate(temp_parsed_rule):
```

Comment:
Could you maybe flatten the if-statements in this loop? That way it is a bit easier to understand. A suggestion cold look like this:

```python
  for segment_idx, segment in enumerate(temp_parsed_rule):
      if isinstance(segment, (Exists, Always)):
          skipped_counter += 1
          continue
      exists_filter = RuleParser._get_exists_filter(segment)
      if exists_filter is None:
          skipped_counter += 1
          continue
      if exists_filter in filter_expression:
          skipped_counter += 1
          continue
      filter_expression.insert(segment_idx * 2 - skipped_counter, exists_filter)
```

Reply:
I've changed it.